tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/statements/ExportStatementsView.java

```diff
@@ -24,42 +25,40 @@
 import org.eclipse.rdf4j.http.server.repository.RepositoryInterceptor;
 import org.eclipse.rdf4j.model.IRI;
 import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
 import org.eclipse.rdf4j.model.Value;
 import org.eclipse.rdf4j.repository.RepositoryConnection;
 import org.eclipse.rdf4j.repository.RepositoryException;
 import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.RDFHandler;
 import org.eclipse.rdf4j.rio.RDFHandlerException;
 import org.eclipse.rdf4j.rio.RDFWriter;
 import org.eclipse.rdf4j.rio.RDFWriterFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.web.servlet.View;
 
 /**
- * View used to export statements. Renders the statements as RDF using a serialization specified using a parameter or
- * Accept header.
+ * Streams statements as RDF in the format requested by the client.
  *
  * @author Herko ter Horst
  */
 public class ExportStatementsView implements View {
 
 	public static final String SUBJECT_KEY = "subject";
-
 	public static final String PREDICATE_KEY = "predicate";
-
 	public static final String OBJECT_KEY = "object";
-
 	public static final String CONTEXTS_KEY = "contexts";
-
 	public static final String USE_INFERENCING_KEY = "useInferencing";
-
 	public static final String CONNECTION_KEY = "connection";
-
 	public static final String TRANSACTION_ID_KEY = "transactionID";
-
 	public static final String FACTORY_KEY = "factory";
-
 	public static final String HEADERS_ONLY = "headersOnly";
 
 	private static final ExportStatementsView INSTANCE = new ExportStatementsView();
+	public static final int MAX_NUMBER_OF_STATEMENTS_WHEN_TESTING_FOR_POSSIBLE_EXCEPTIONS = 1024;
```

Comment:
Make this configurable through system property, and also support -1 as infinity.